core/optaplanner-core-impl/src/main/java/org/optaplanner/core/api/domain/variable/VariableListener.java

```diff
@@ -12,6 +12,17 @@
  */
 public interface VariableListener<Solution_, Entity_> extends AbstractVariableListener<Solution_, Entity_> {
 
+    /**
+     * When set to {@code true}, this has a slight performance loss in Planner.
+     * When set to {@code false}, it's often easier to make the listener implementation correct and fast.
+     *
+     * @return true to guarantee that each of the before/after methods will only be called once per entity instance
+     *         per operation type (add, change or remove).
+     */
+    default boolean requiresUniqueEntityEvents() {
```

Comment:
How about we create an issue to investigate if we can @Deprecate this method? If always acting as false is doable for all our examples/quickstarts. It's a needless configuration option that no one needs.

Reply:
https://issues.redhat.com/browse/PLANNER-2809